docs/psf_spec/high_level_overview.rst

```diff
@@ -0,0 +1,244 @@
+system -- optics, CCD, etc. -- that created the image. See `below <PSF Model_>`_ for more
+details on the PSF model, and the :doc:`fitter <fitter>` documentation for further
+information on the ``Fitter`` instance and its properties.
+
+
+.. _PSF Model:
+
+PSF Model
+^^^^^^^^^
+Perhaps the most important aspect of PSF fitting, the ``PSF Model`` describes the distribution
+of light falling on a given pixel from a source at some position with some total flux, fully
+describing the effects of telescope optics, quantized CCD pixels, etc. In this respect it is
+more properly described as a PRF than a PSF -- see the :doc:`documentation <psf_model>` for
+more details -- but we conflate the two terms here freely. Similar to the ``Fitter``, it is
+a separate callable class detailing a specific PSF response. This can be some analytical
+function, such as `~photutils.psf.IntegratedGaussianPRF`, or an empirical description of the
+PSF, such as that implemented by the effective PSF functionality (see :ref:`build-epsf` for
+details). The model must describe a two-dimensional distribution, and must return the total
+flux of the source and its sub-pixel position via the ``evaluate`` function, accepted by the
+Fitter_ when minimizing the fit between the model and the data.
+
+
+.. _noise:
+
+Noise Description
+^^^^^^^^^^^^^^^^^
+The ``noise`` block within the PSF fitting process is, in much the same way as the ``Fitter``
+block, a separate ``callable`` function mapping the relationship between the pixel values in
+the input ``image`` and the corresponding uncertainty -- in units of standard deviations -- of
+the pixel. It must therefore accept a two-dimensional array and return an array of the same
+shape as ``image``. Alternatively it can be overloaded to indicate that the uncertainty array
+was passed along with ``image`` via an `~astropy.nddata.NDData` instance. A more detailed
+description of this block is available :doc:`here <noise_data>`.
+
+
+.. _Background Estimator:
+
+Background Estimator
+^^^^^^^^^^^^^^^^^^^^
+The ``image`` on which source extraction and evaluation is to be performed is not necessarily
+assumed to have had any pre-processing applied to it, and thus it may be necessary to account
+for the counts of the image background when handling PSF fitting. This block handles
+the determination of the background levels, avoiding high count peaks to calculate the typical
+count of an otherwise empty pixel. The block must therefore accepts the two-dimensional ``image``
+and returns either a single value -- the background count of the entire image, as calculated
+by the criteria of the specific block implementation -- or a 2D array of values, allowing for
+the possibility of varying background counts across the image. More information on this block
+can be found :doc:`here <background_estimator>`.
+
+
+.. _Culler and Ender:
+
+Culler and Ender
+^^^^^^^^^^^^^^^^
+Only implemented in iterative fitting classes, part of the extension to
```

Comment:
I don't understand the "part of the extension to" part... partly because `BasicPSFPhotometry` isn't really defined here (see my general comment)